src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountingExtractCollectorBatchBuilder.java

```diff
@@ -282,44 +321,147 @@ protected String buildTransactionDescription(ConcurStandardAccountingExtractDeta
-        }
+    protected void reportCashAdvance(ConcurStandardAccountingExtractDetailLine saeLine) {
+        updateStatisticFromLineData(ConcurStandardAccountingExtractBatchReportData::getCashAdvancesRelatedToExpenseReports, saeLine);
+    }
+
+    protected void reportCorporateCardPayment(ConcurStandardAccountingExtractDetailLine saeLine) {
+        updateStatisticFromLineData(ConcurStandardAccountingExtractBatchReportData::getExpensesPaidOnCorporateCard, saeLine);
+    }
+
+    protected void reportBypassOfLineWithPseudoPaymentCode(ConcurStandardAccountingExtractDetailLine saeLine) {
+        updateStatisticFromLineData(ConcurStandardAccountingExtractBatchReportData::getTransactionsBypassed, saeLine);
+    }
+
+    protected void updateStatisticFromLineData(
+            Function<ConcurStandardAccountingExtractBatchReportData,ConcurBatchReportSummaryItem> summaryItemGetter,
+            ConcurStandardAccountingExtractDetailLine saeLine) {
+        ConcurBatchReportSummaryItem summaryItem = summaryItemGetter.apply(reportData);
+        summaryItem.setRecordCount(summaryItem.getRecordCount() + 1);
+        summaryItem.setDollarAmount(summaryItem.getDollarAmount().add(saeLine.getJournalAmount().abs()));
+    }
+
+    protected void reportPendingClientLine(ConcurStandardAccountingExtractDetailLine saeLine) {
+        ConcurBatchReportMissingObjectCodeItem pendingClientItem = new ConcurBatchReportMissingObjectCodeItem(
+                saeLine.getReportId(),
+                saeLine.getEmployeeId(),
+                saeLine.getEmployeeLastName(),
+                saeLine.getEmployeeFirstName(),
+                saeLine.getEmployeeMiddleInitital(),
+                "Line has the \"Pending Client\" object code",
+                saeLine.getPolicy(),
+                saeLine.getExpenseType());
+        reportData.addPendingClientObjectCodeLine(pendingClientItem);
+    }
+
+    protected void reportUnprocessedLine(ConcurStandardAccountingExtractDetailLine saeLine, String errorMessage) {
+        ConcurBatchReportLineValidationErrorItem errorItem = new ConcurBatchReportLineValidationErrorItem(
+                saeLine.getReportId(),
+                saeLine.getEmployeeId(),
+                saeLine.getEmployeeLastName(),
+                saeLine.getEmployeeFirstName(),
+                saeLine.getEmployeeMiddleInitital(),
+                errorMessage);
+        reportData.addValidationErrorFileLine(errorItem);
     }
 
-    protected void addLineAmountToOriginEntry(ConcurStandardAccountingExtractDetailLine saeLine, OriginEntryFull originEntry) {
-        String debitCreditCode = getGLDebitCreditCode(saeLine.getJounalDebitCredit());
-        KualiDecimal amount = saeLine.getJournalAmount();
+    /**
+     * Internal helper class for encapsulating related SAE transactions,
+     * and for generating Collector OriginEntryFull lines from them.
+     */
+    protected class DetailLineGroup {

```

Comment:
Making it private would make overriding more difficult, but that would only be a problem if we contributed this back to the foundation. I may be refactoring this to make the internal class a standalone one instead.